Merge updated base python buildpack code

ninapavlich opened the pull request:
Hello! It would be super to update this repo to include the latest Python buildpack code.

I'm not sure how complicated this merge would be, but if it's easy then it would be great to do!

kennethreitz committed:
pypy3-5.8.0 (#414)

kennethreitz committed:
Python 3 default for buildpack (#409)

* update default python to 3.6.1

* v110

* nltk python2

kennethreitz committed:
Update README.md (#415)

joshfriend committed:
Add CPython 3.6.2 (#422)

kennethreitz committed:
Python 3.6.2 (#423)

* pypy3-5.8.0

* updated cpython to 3.6.2

* update README to reflect new Python version

edmorley committed:
Exclude Python's test suite from generated Python archives (#425)

This copies the official Docker Python images approach of stripping
the Python test suite after building from source.

For Python 3.6.2 this reduces the output from 184MB to 121MB and
removes the following directories:
```
.heroku/python/lib/python3.6/tkinter/test
.heroku/python/lib/python3.6/ctypes/test
.heroku/python/lib/python3.6/unittest/test
.heroku/python/lib/python3.6/distutils/tests
.heroku/python/lib/python3.6/sqlite3/test
.heroku/python/lib/python3.6/lib2to3/tests
.heroku/python/lib/python3.6/test
```

For Python 2.7.13 this reduces the output from 127MB to 91MB and
removes the following directories:
```
.heroku/python/lib/python2.7/ctypes/test
.heroku/python/lib/python2.7/unittest/test
.heroku/python/lib/python2.7/distutils/tests
.heroku/python/lib/python2.7/bsddb/test
.heroku/python/lib/python2.7/lib-tk/test
.heroku/python/lib/python2.7/sqlite3/test
.heroku/python/lib/python2.7/lib2to3/tests
.heroku/python/lib/python2.7/json/tests
.heroku/python/lib/python2.7/email/test
.heroku/python/lib/python2.7/test
```

Fixes #424.

joshfriend committed:
Add pypy2-5.8.0 (#430)

kennethreitz committed:
fixed the bug for pypy-5.8.0 (#431)

codingjoe committed:
Update GDAL version to 1.11.5 (#432)

* Update GDAL version to 1.11.5

Update GDAL to latest bugfix release, see
http://trac.osgeo.org/gdal/wiki/Release/1.11.5-News

* Use pushd and popd in favor of cd

kennethreitz committed:
Geos tests (#437)

* fixed the bug for pypy-5.8.0

* cleanup makefile

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* allow builds to happen when capitalization is used

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* restore behavior

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* make makefile default to heroku-16

* test geos libraries

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Shellcheck Compliance (#438)

* fixed the bug for pypy-5.8.0

* enable shellcheck in makefile

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* make compile script shellcheck compatible

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* let's see what this says

* fix pip installation

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck detect script

* shellcheck release

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck test-compile

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck warnings/utils

* updated makefile

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck collectstatic

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck cryptography

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck eggpath fixes

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck gdal

* shellcheck goes-libs

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck mercurial

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* nltk shellcheck

* shellcheck pip-install

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck pip-uninstall

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck pipenv

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck pylibmc

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* shellcheck python

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* update makefile

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* final update to makefile

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* fix tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* actually fix the tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* update pipenv-python-version

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* requested fixes

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* update spellcheck sources

* attempted fix for shell globbing

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* further attempted fix

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* switch nltk styling

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* no more need for temp app for installing pip

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* attempt at staged build for shellcheck

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* attempted fix of travis file

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* reorder travis file

* add shellcheck to install

* make things clearer in travis file

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* better travis file

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* attempt at better travis file

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* better travis file

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

edmorley committed:
Improve UX when clearing cache due to the stack changing (#442)

* Test that the cache is invalidated when the stack changes

* Improve UX when clearing cache due to the stack changing

Now outputs a message informing that the cache was cleared, and
clears the cache first to avoid a redundant message about removing
an old Python version.

edmorley committed:
Speed up the multi-stage Travis run (#443)

Reduces the impact on Travis runtime caused by #438, by:
* Making the shellcheck job run on the container infra (`sudo: false`)
  which has faster boot times.
* Running the heroku-16 and cedar-14 stack tests in parallel, since
  previously they were separate stages (which are run in series by
  design).

Reduces Travis end-to-end time from ~10 minutes to ~6 minutes.

jmorrell committed:
Remove leading zeros from `WEB_CONCURRENCY` (#446)

* Remove leading zeros from `WEB_CONCURRENCY`

* fix DYNO_RAM (from node buildpack)

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

LegoStormtroopr committed:
Allow environment variables during pip install, fixes #416 (#417)

* Allow environment variables during pip install, fixes #416

* shift sub-env command, fix typo

* bash check thingy fixed

kennethreitz committed:
Changelog (#454)

* fixed the bug for pypy-5.8.0

* changelog entry

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Grep bug fix (#456)

* fixed the bug for pypy-5.8.0

* changelog entry

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* use std-lib sub-env

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* let's see if this cleans things up

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try v7 of stdlib

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* use new sub_env

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* set_env

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* set_default_env

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* v8 of stdlib

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* v112

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops (merge bug)

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* final merge error resolve

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* final final fix for merge conflict

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Pipenv Updates (#458)

* fixed the bug for pypy-5.8.0

* update pipenv support

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Update local commits (#459)

* fixed the bug for pypy-5.8.0

* update changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

edmorley committed:
NLTK support: Fix passing of multiple corpora identifiers (#460)

* NLTK support: Update test to use multiple corpora

So that the incorrect handling of multiple IDs seen in #444 would
have been caught.

Also switches to some of the smaller corpora, to reduce time spent
downloading during tests (see sizes on http://www.nltk.org/nltk_data/).

* NLTK support: Fix passing of multiple corpora identifiers

As part of fixing the shellcheck warnigns in #438, double quotes had
been placed around `$nltk_packages` passed to the `nltk.downloader`,
which causes multiple identifiers to be treated as though it were just
one identifier that contains spaces.

The docs for the shellcheck warning in question recommend using arrays
if the intended behaviour really is to split on spaces:
https://github.com/koalaman/shellcheck/wiki/SC2086#exceptions

As such, `readarray` has been used, which is present in bash >=4.
The `[*]` array form is used in the log message, to prevent shellcheck
warning SC2145, whereas `[@]` is used when passed to `nltk.downloader`
to ensure the array elements are unpacked as required.

Note: Both before and after this fix, using anything but unix line
endings in `nltk.txt` will also cause breakage.

edmorley committed:
Tests: Make assertCaptured test failures easier to debug (#461)

Previously the contents of the file being searched was not output,
making it hard to know how to fix a test so that it would pass.

Before:
```
$ make test-heroku-16
...
testCffi
ASSERT:Expected </tmp/shunit.ayVGAO/tmp/output.6gMs/stdout> to contain <ZZZZZZZ>
testPylibmc
```

After:
```
$ make test-heroku-16
...
testCffi
ASSERT:Expected </tmp/shunit.FXWeuI/tmp/output.7Vfx/stdout> to contain <ZZZZZZZ>
 !     Warning: Your application is missing a Procfile. This file tells Heroku how to run your application.
 !     Learn more: https://devcenter.heroku.com/articles/procfile
-----> Installing python-3.6.2
-----> Installing pip
-----> Installing requirements with pip
       Collecting cffi (from -r /tmp/testBoLG0/requirements.txt (line 1))
         Downloading cffi-1.10.0-cp36-cp36m-manylinux1_x86_64.whl (406kB)
       Collecting pycparser (from cffi->-r /tmp/testBoLG0/requirements.txt (line 1))
         Downloading pycparser-2.18.tar.gz (245kB)
       Installing collected packages: pycparser, cffi
         Running setup.py install for pycparser: started
           Running setup.py install for pycparser: finished with status 'done'
       Successfully installed cffi-1.10.0 pycparser-2.18

testPylibmc
```

kennethreitz committed:
Update README.md (#462)

* Update README.md

* Update README.md

* Update README.md

kennethreitz committed:
Update README.md (#463)

kennethreitz committed:
Update README.md (#464)

kennethreitz committed:
Blacklist for old apps (#466)

* fixed the bug for pypy-5.8.0

* update changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* blacklist for old apps

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* only blacklist pythonhome/path

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* updated changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Undo sub_env for pip install, it was a terrible idea (#467)

* fixed the bug for pypy-5.8.0

* update changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* blacklist for old apps

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* only blacklist pythonhome/path

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* updated changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* be more agressive with blacklisting

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* sub_env was a terrible idea

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* changelog update

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

edmorley committed:
Correct changelog entries for v108, v109 and v110 (#469)

Found by diffing the published buildpack archives and looking at the
git log to see what changed. It looks like the v108 entry was
accidentally added in #405, when the release being published there
was actually v107. The setuptools update mentioned there never
happened, since the `SETUPTOOLS_VERSION` variable was found to be
unused and instead removed.

edmorley committed:
Skip vendoring libffi/libmemcached on Heroku-16 (#465)

* Add a test for the cryptography (cffi) compile step

Since it's currently untested.

* Skip vendoring libffi/libmemcached on Heroku-16

Unlike for Cedar-14, Heroku-16 comes with these packages pre-installed:
https://github.com/heroku/stack-images/blob/5a341970cfc1f201014262ad64c3b3e47514f663/heroku-16/installed-packages.txt#L111
https://github.com/heroku/stack-images/blob/5a341970cfc1f201014262ad64c3b3e47514f663/heroku-16/installed-packages.txt#L172

As such the build scripts had already been made a no-op on Heroku-16:
https://github.com/heroku/heroku-buildpack-python/blob/fedae5ceda5a42f594012f911c3808dd5dc6fd9e/builds/libraries/vendor/libffi#L6-L9
https://github.com/heroku/heroku-buildpack-python/blob/fedae5ceda5a42f594012f911c3808dd5dc6fd9e/builds/libraries/vendor/libmemcache#L6-L9

...meaning the Heroku-16 archives for them on S3 contain zero files.

However until now, the buildpack was still unnecessarily downloading
and extracting these empty archives - and not just on the first compile
(like on cedar-14), but every compile since the directory check will
never succeed.

Commit:
Adding step to copy libjasper to vendor lib directory so that libgdal will load (#449)

kennethreitz committed:
update the changelog (#470)

kennethreitz committed:
Improvements (#471)

* only do this on heroku-16

* history

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* code comment

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

codingjoe committed:
Update geos to version 3.6.2 (#450)

codingjoe committed:
Update proj.4 to version 4.9.3 (#451)

codingjoe committed:
Update gdal to version 2.2.1 (#452)

kennethreitz committed:
Pipenv improvements (#474)

* improvements to pipenv

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* capitalize

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Upgrades 🔝 (#475)

* fixed the bug for pypy-5.8.0

* changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* python 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* pipfile > requirements.txt

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* check for python_full_version too

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* warn when using an older version of python

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* requirements for anaconda buildpack

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* remove hashes for conda buildpack

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* improvements to pipenv python version detection

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* echo not puts

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* learn more

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* there we go

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* unsupported

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* consistiency

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* be more specific

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* capitalize Pipfile.lock

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* attempt to force color

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* interactive

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* interactive

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* diagnose

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try without -l

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* confirmed env working

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* just -c

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* use latest, to debug

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* no more bash

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try …

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* don't use the git version of pipenv

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* next version

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* comment out force color bits

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* readme

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* more tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* more tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* ellipsis

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
pipenv --deploy (#476)

* fixed the bug for pypy-5.8.0

* changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* python 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* pipfile > requirements.txt

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* check for python_full_version too

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* warn when using an older version of python

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* requirements for anaconda buildpack

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* remove hashes for conda buildpack

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* improvements to pipenv python version detection

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* echo not puts

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* learn more

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* there we go

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* unsupported

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* consistiency

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* be more specific

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* capitalize Pipfile.lock

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* attempt to force color

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* interactive

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* interactive

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* diagnose

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try without -l

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* confirmed env working

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* just -c

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* use latest, to debug

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* no more bash

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try …

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* don't use the git version of pipenv

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* next version

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* comment out force color bits

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* readme

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* more tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* more tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* ellipsis

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* pipenv --deploy

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* full version lockfile

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

joshfriend committed:
Add CPython 3.6.3 (#479)

codingjoe committed:
Fix #398 -- Set explict GDAL and GEOS location (#478)

Set GDAL and GEOS library locaiton explicitly in environment
variables:

*   GDAL_LIBRARY_PATH "/app/.heroku/vendor/lib/libgdal.so"
*   GEOS_LIBRARY_PATH "/app/.heroku/vendor/lib/libgeos_c.so"

Django has to settings with the same name. The setup now works as
described here:
https://devcenter.heroku.com/articles/postgis#geodjango-setup

joshfriend committed:
Update default python to 3.6.3 (#480)

kennethreitz committed:
update changelog (#486)

* fixed the bug for pypy-5.8.0

* changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* python 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* pipfile > requirements.txt

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* check for python_full_version too

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* warn when using an older version of python

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* requirements for anaconda buildpack

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* remove hashes for conda buildpack

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* improvements to pipenv python version detection

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* echo not puts

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* learn more

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* there we go

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* cleanups

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* unsupported

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* consistiency

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* be more specific

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* capitalize Pipfile.lock

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* attempt to force color

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try this

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* interactive

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* interactive

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* diagnose

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try without -l

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* confirmed env working

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* just -c

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* use latest, to debug

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* no more bash

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* try …

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* don't use the git version of pipenv

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* oops

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* next version

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* 2.7.14

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* comment out force color bits

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* readme

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* more tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* more tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* ellipsis

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* Merge branch 'master' of github.com:heroku/heroku-buildpack-python

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Make gunicorn default access log to stdout (#490)

* make gunicorn default access log to stdout

* Update python.gunicorn.sh

Androbin committed:
Fix some Bash issues (#491)

kennethreitz committed:
Mcount pipenv and pip (#492)

* mcount pipenv

* more mcount for pip

* shellcheck compliance

* fix typo

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

joshfriend committed:
Add CPython 3.6.4 (#495)

kennethreitz committed:
latest python version (3.6.4) (#496)

Update buildpack to latest version of Python.

kennethreitz committed:
Update python.gunicorn.sh (#628)

kennethreitz committed:
Changelog Updates (#630)

* changelog notice

* v123

* v123

kennethreitz committed:
Pipenv development dependencies (ci) (#633)

* initial stab at pipenv --dev support

* test installation output

* locale fix

* refactor

* fix typo

* fix typo

* ellipsis

* polish

kennethreitz committed:
Update README.md (#642)

Photonios committed:
Account for the root not being /app (#638)

p-himik committed:
Add support for PIP_EXTRA_INDEX_URL (#639)

* Add support for PIP_EXTRA_INDEX_URL

* Add support for PIP_EXTRA_INDEX_URL for Pipenv

josegonzalez committed:
feat: allow stack url overrides using BUILDPACK_VENDOR_URL env var (#643)

This is useful if you'd like to customize python binaries without forking the entire buildpack.

Similar functionality was implemented in https://github.com/heroku/heroku-buildpack-ruby/pull/238

jxltom committed:
Fix PYTHONPATH is not set when running collectstatic (#637)

* Fix PYTHONPATH is not set when running collectstatic

* update changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* Update CHANGELOG.md

kennethreitz committed:
Update pipenv-python-version to latest python 3.6 (#644)

* Update pipenv-python-version

* use latest versions in pipenv script

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* Update pipenv-python-version

* 3.6.4

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Update README.md (#649)

kennethreitz committed:
Pipenv uninstall, and other improvements (#650)

kennethreitz committed:
fix changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Update CHANGELOG.md

kennethreitz committed:
Update CHANGELOG.md

kennethreitz committed:
Update CHANGELOG.md

kennethreitz committed:
Update CHANGELOG.md

kennethreitz committed:
remove cryptogrpahy step, as they ship wheels now

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Merge branch 'master' of github.com:heroku/heroku-buildpack-python

kennethreitz committed:
cleanup

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
set -e

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
reorder things

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
don't suppress error messages

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
don't expect there to be a requirements.txt

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
always use pipenv if it's there

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
fix the problem

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
don't print extra information

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
also skip pip install

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
source pipenv step

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
no longer check cryptogrpahy step

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
fix mercurial step

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
bugfixes

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Metrics (#657)

* metrics

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* using pipenv anyway

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* instrument bad failures

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* if not

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
default to user python path for collectstatic runs (#655)

* default to user python path for collectstatic runs

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* bugfixes

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* fix

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
don't skip Pipenv installation if there are git deps (#656)

kennethreitz committed:
pin to pipenv==11.6.1

kennethreitz committed:
Update CHANGELOG.md

kennethreitz committed:
Update pipenv

kennethreitz committed:
collectstatic metrics (#660)

zyv committed:
Output pipenv version in steps (#662)

* Since pipenv is being pinned now, output pinned version in steps

Signed-off-by: Yury V. Zaytsev <yury.zaytsev@moneymeets.com>

* Update README.md example output to match new reality

Signed-off-by: Yury V. Zaytsev <yury.zaytsev@moneymeets.com>

* Update pipenv

kennethreitz committed:
pipenv

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
upgrade pip

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
don't upgrade pip

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
cleanup pip upgrade step

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
9.0.2

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
pip

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
9.0.2

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
fix bug with pip

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
update pip installation method, to include setuptools

kennethreitz committed:
improve pip installation

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
codeowners (#668)

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Pip frozen version (#683)

* freeze pip version, upon installation

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* use PIP_UPDATE environment variable

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* disable shell checking (for now)

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* --disable-pip-version-check

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
changelog

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

joshfriend committed:
Add CPython 3.6.5 (#676)

fixes #674

kennethreitz committed:
update metrics

kennethreitz committed:
test git egg installation (#666)

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

hone committed:
only need to build libmemcache dev for cedar-14 (#690)

sigmavirus24 committed:
Add Python 2.7.15 to the list of runtimes (#692)

* Add Python 2.7.15 to the list of runtimes

Closes #691

* Update the default Python 2 to 2.7.15 everywhere

kennethreitz committed:
Update README.md

kennethreitz committed:
changelog entry

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
update latest version of python

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
lots of comments

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Update compile

kennethreitz committed:
fix tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
Merge branch 'master' of github.com:heroku/heroku-buildpack-python

kennethreitz committed:
Python 3.6.5 (#695)

* 3.6.5

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

* fix tests

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

kennethreitz committed:
buildpack.toml

Signed-off-by: Kenneth Reitz <me@kennethreitz.org>

pdamodara committed:
Update pipenv-to-pip to read lockfiles properly

Use the appropriate encoding to read a Pipfile.lock so we can
properly load the data.

Closes #678

sigmavirus24 committed:
Let's test re-enabling shellcheck in CI

shellcheck is a truly valuable utility. Unfortunately installing the
packages from debian-sid introduces a conflict. Let's try another
suggestion from: https://github.com/koalaman/shellcheck/wiki/TravisCI

Refs
https://github.com/heroku/heroku-buildpack-python/commit/a0275888a26b8f0132c7bb9fe3bf194b5c4fab1f

sigmavirus24 committed:
Fix-up SC2219 errors in our shell scripts

Caught these with the re-added shellcheck linting.

sigmavirus24 committed:
Merge pull request #699 from heroku/reenable-shellcheck

Let's test re-enabling shellcheck in CI

Commit:
Bump pipenv version

Commit:
Fix shellcheck linting issues

sigmavirus24 committed:
Fix new shellcheck warnings breaking periodic CI

New warnings were found in our periodic (CRON) CI jobs.

Fixes https://travis-ci.org/heroku/heroku-buildpack-python/jobs/383404864

sigmavirus24 committed:
Merge pull request #703 from heroku/fix-new-shellcheck-warnings

Fix new shellcheck warnings breaking periodic CI

sigmavirus24 committed:
Merge branch 'master' into bump-pipenv-version

sigmavirus24 committed:
Merge pull request #701 from allyjweir/bump-pipenv-version

Bump pipenv version

KevinBrolly committed:
Updated README for python 3.6.5 and CHANGELOG for v135

sigmavirus24 committed:
Merge pull request #710 from KevinBrolly/master

Updated README for python 3.6.5 and CHANGELOG for v135

hone committed:
setup libsqlite3-dev and sqlite3 binary to match stack's libsqlite3-0

With inspiration from @KevinBrolly, this patch uses the stack image
SQLite3 package but also still providing the dev headers and binary that
users may still be using today. The benefit is that we won't need to
rebuild all the python binaries for this to take affect. We can just
stop shipping SQLite3 from future binaries. In addition, we don't need
to worry about what version and when to update SQLite3 and maintaining
the packages ourselves.

This also includes updates to Python 2.7.15 and Python 3.6.6 so they can
rebuilt with the stack image dev headers instead of building our own
vendored SQLite3.

sigmavirus24 committed:
Build Python 3.6.6 and 3.7.0

Add runtime build recipes for our newly released versions

Closes gh-720
Closes gh-721

sigmavirus24 committed:
Update defaults for Python 3 apps on Heroku

Also update our documentation and CHANGELOG for this version of the
buildpack.

sigmavirus24 committed:
Merge pull request #722 from heroku/python-3.7-and-3.6.6

Build Python 3.6.6 and 3.7.0

ianyh commented:
Cool! I haven’t touched this in forever, though, so I’m not sure about the conflicts. I’d be happy to give you commit access to this repo if you want.

sigmavirus24 committed:
Steal the java buildpack's release script

This will make releasing new versions easier for us.

Closes gh-723

sigmavirus24 committed:
Merge pull request #724 from heroku/publish-automation

Steal the java buildpack's release script

sigmavirus24 committed:
Prevent Python 3.7 from being unsupported

Python 3.7.0 is supported but not preferred given how new it is. As a
result, we don't want it to be the default, but we also don't want users
to be confused when upgrading to it.

Closes gh-728

sigmavirus24 committed:
Add heroku-18 to our Travis CI config

- Add stage to Travis CI config and update tests.sh script to recognize
  it

- Update tests to assert there is no Python 2 on Heroku-18

- Update nltk fixture to use Python 3.6 so we can test it on all stacks

Closes gh-730

sigmavirus24 committed:
Merge pull request #731 from heroku/heroku-18-ci

Add heroku-18 to our Travis CI config

sigmavirus24 committed:
Be clearer in our warnings about which Py3 we mean

sigmavirus24 committed:
Merge branch 'master' into bug/728

sigmavirus24 committed:
Merge pull request #729 from heroku/bug/728

Prevent Python 3.7 from being unsupported

hone committed:
add Python version operators

hone committed:
v137 changelog

hone committed:
Merge pull request #733 from heroku/v137

v137 changelog

hone committed:
Merge pull request #713 from hone/sqlite3

Update SQLite3

CaseyFaist committed:
v 138 changelog

CaseyFaist committed:
Merge pull request #736 from heroku/v-138

v 138 changelog

schneems committed:
Integration tests with Hatchet

schneems committed:
Add retry gem

CaseyFaist committed:
Update indentation levels and messaging

Fix indentation levels

more helpful messages about supported versions

CaseyFaist committed:
add case for older python 3 install attempts

CaseyFaist committed:
reword warning, add doc link

CaseyFaist committed:
changelog update

CaseyFaist committed:
setup travis rspecting

CaseyFaist committed:
correct travisyaml indenting

CaseyFaist committed:
add rake to gemfile

CaseyFaist committed:
add Rakefile

CaseyFaist committed:
use bundle syntax to rspec

edmorley committed:
Add a Pull Request template

To try and help prevent the notification noise from PRs being opened
against upstream by mistake.

CaseyFaist committed:
Merge pull request #744 from edmorley/add-pr-template

Add a Pull Request template

CaseyFaist committed:
update tests to expect success for all 2.7.15 builds

CaseyFaist committed:
typo fix

jmorrell committed:
Start adding dates to changelog entries

It's occasionally useful to see at a glance when releases were pushed to understand what release might have caused a customer's issues. This back-fills dates for the last few months of releases and sets a precedent that can be followed in the future.

CaseyFaist committed:
Merge pull request #745 from heroku/fix-27-tests

update buildpack tests to expect success for all 2.7.15 builds

CaseyFaist committed:
Merge pull request #746 from heroku/changelog-dates

Start adding dates to changelog entries

CaseyFaist committed:
add missing newline

CaseyFaist committed:
Merge pull request #742 from heroku/hatchet/yay

Hatchet/yay

CaseyFaist committed:
Merge pull request #741 from heroku/fix-python-check

Fix Python 3.6 message when using Python 3.7

CaseyFaist committed:
fix travis dependency blocker

CaseyFaist committed:
rememer to remove toolbelt

CaseyFaist committed:
move heroku install to before_install

CaseyFaist committed:
add shebang and update file permissions

CaseyFaist committed:
add sudo to travis install

CaseyFaist committed:
add more unblocking setup

CaseyFaist committed:
Merge pull request #759 from heroku/update-travis

Update travis setup for Hatchet

CaseyFaist committed:
add SLUGIFY_USES_TEXT_UNIDECODE support

CaseyFaist committed:
add test/run case

CaseyFaist committed:
need to pass env var as export in test

CaseyFaist committed:
Merge pull request #760 from heroku/add-airflow-110-support

Add Apache Airflow 1.10 Support

duanehutchins committed:
Updated HATCHET_BUILDPACK_BASE to use TRAVIS_REPO_SLUG

CaseyFaist committed:
pin pip version used by pipenv

CaseyFaist committed:
changelog update

CaseyFaist committed:
Merge pull request #769 from heroku/pip-pipenv-pin

Pin pip version used by pipenv

CaseyFaist committed:
add new runtimes and test repos

CaseyFaist committed:
add tests for new runtimes

CaseyFaist committed:
adjust sqlcheck

CaseyFaist committed:
fix extra slash in python install location

josegonzalez committed:
fix: drop broken version check

Due to how the version checks work - via sorting, not actual comparison - this previously resulted in _always_ installing sqlite3, even though it was already bundled for lower versions of python. The second version check also encompasses 3.7.0+, so there is no need to respecify the check.

CaseyFaist committed:
update bob dependency to point to internal bob-builder fork

CaseyFaist committed:
remove comment

CaseyFaist committed:
delete me later

CaseyFaist committed:
use released bob package

CaseyFaist committed:
delete tester runtime.txt

Added this for local testing because bob requires 2.7, should not be merged to master

CaseyFaist committed:
Merge pull request #777 from heroku/use-heroku-bob

update bob dependency to point to internal bob-builder fork

CaseyFaist committed:
update changelog

CaseyFaist committed:
update tests to pass, add 3.4

CaseyFaist committed:
use text variable as variable in puts-warn

CaseyFaist committed:
add bash pre-commit hook and correct shellcheck warnings

CaseyFaist committed:
add python 3.4 detection

CaseyFaist committed:
add missing vars for python step

CaseyFaist committed:
specify python 2.7

CaseyFaist committed:
update changelog

CaseyFaist committed:
add warning detection and per-stack checks

CaseyFaist committed:
update tests for missed stack issues

CaseyFaist committed:
Merge pull request #772 from heroku/v-145

V 145

CaseyFaist committed:
update to newer hatchet integration

CaseyFaist committed:
add HATCHET_DEPLOY_STRATEGY env var

CaseyFaist committed:
Merge pull request #781 from heroku/update-hatchet-setup

Update to reflect Hatchet updates

CaseyFaist committed:
add next runtimes

CaseyFaist committed:
update test fixtures for new python runtimes

CaseyFaist committed:
bump default 3.6 version

CaseyFaist committed:
update 3.7 tests typo

CaseyFaist committed:
update latest vars in compile

CaseyFaist committed:
update default in pipenv test

CaseyFaist committed:
update hatchet expected version

CaseyFaist committed:
add test for 2.7 pipenv install

CaseyFaist committed:
update latest python 2 in pipenv-version step

CaseyFaist committed:
remove template runtimes to cut down on scope creep

hunterloftis committed:
add heroku-18 to test targets and document running tests

CaseyFaist committed:
Merge pull request #782 from heroku/next-release

Test updates

hunterloftis committed:
Merge branch 'master' of github.com:heroku/heroku-buildpack-python into document-tests

CaseyFaist committed:
Merge pull request #783 from heroku/document-tests

Document tests

CaseyFaist committed:
update changelog

CaseyFaist committed:
update release date

CaseyFaist committed:
update readme

CaseyFaist committed:
Merge pull request #784 from heroku/update-changelog

update changelog

CaseyFaist committed:
Merge pull request #785 from heroku/update-readme

Update readme

CaseyFaist committed:
changelog and test update

CaseyFaist committed:
new runtimes and travis test updates

CaseyFaist committed:
correct date mixup + update changelog

CaseyFaist committed:
Merge pull request #797 from heroku/v-147

V 147

CaseyFaist committed:
track usage of older feature steps (#802)

* track usage of older feature steps

* remove typo whitespace

* make buildvars consistently cased

CaseyFaist committed:
update changelog (#803)

dzuelke committed:
HOTFIX for cedar-14 python builds (#805)

CaseyFaist committed:
Python 2.7.16 (#808)

* new recipe for new runtime

* add new runtime formula

* wrangle tests into submission

CaseyFaist committed:
changelog update for v149 (#806)

CaseyFaist committed:
Python release prep (#809)

* new recipe for new runtime

* add new runtime formula

* add test updates for new runtime release

* wrangle tests into submission

* update tests to use default_pythons

* delete commented code

CaseyFaist committed:
changelog 150 update (#810)

CaseyFaist committed:
Release prep 151 (#813)

* new runtimes

* update default latest

CaseyFaist committed:
add new python and update default

CaseyFaist committed:
fix test blindspot

CaseyFaist committed:
update CHANGELOG

CaseyFaist committed:
changelog date corrected

CaseyFaist committed:
remove test case for deprecated pythons on deprecated stack

CaseyFaist committed:
Merge pull request #814 from heroku/v-152

Python 3.7.3 release

dzuelke committed:
HOTFIX heroku-16 builds with sqlite3

upstream version has changed to 3.11.0-1ubuntu1.2, but local package version is still on 3.11.0-1ubuntu1.1, so the download fails otherwise

CaseyFaist committed:
Merge pull request #826 from heroku/hotfix-sqlite3-heroku-16

HOTFIX heroku-16 builds with sqlite3

dzuelke committed:
CHANGELOG for v153

dzuelke committed:
fix python 3.5.7 formula actually building 3.7.2

dzuelke committed:
avoid service restart prompt during Dockerfile builds

https://unix.stackexchange.com/questions/146283/how-to-prevent-prompt-that-ask-to-restart-services-when-installing-libpq-dev

dzuelke committed:
Merge pull request #827 from heroku/fix-python-3-5-7

Fix Python 3.5.7

josegonzalez committed:
feat: add docs and make target for heroku-18 bob builds

CaseyFaist committed:
Merge pull request #829 from josegonzalez/heroku-18-build

feat: add docs and make target for heroku-18 bob builds
